plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -208,7 +211,7 @@ public InstanceKey saveToSnapshot(String owner) throws MachineException {
 
             docker.removeImage(registry + '/' + repository, false);
 
-            return new DockerInstanceKey(repository, "latest", registry, digest);
+            return new DockerMachineSource(repository).setRegistry(registry).setDigest(digest).setTag(LATEST_TAG);
```

Comment:
I've removed 

but it's funny because here I clearly see this usage
https://github.com/eclipse/che/blob/a8dfb6bf541940c636590119ef2fb45c1fa659c9/wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/MachineSourceImpl.java#L44-56

also this kind of stuff can't be a rule maybe like a current practice but it's not true as I saw code using it. But clearly the with() usage is anti pattern as set() can of course return the instance of the object. It is more "contributor specific"
It avoids two methods and only change the return type. Also with() is clearly more seen with a fluent builder pattern than when there is no associated builder.

I may as well say that with fluent pattern, the usage of 'set' or 'get' is deprecated in favor of a more readable pattern
new MachineSource("type).location("my-location")
vs
new MachineSource("type").setLocation("my-location")
or new MachineSource("type").withLocation("my-location")

this kind of old stuff introduces too much verbosity which is not what tends to be "fluent " so I think it may be a good idea to let contributors pickup what they want